apps/studio/pages/authorize.tsx

```diff
@@ -108,17 +108,24 @@ const APIAuthorizationPage: NextPageWithLayout = () => {
     )
   })
 
-  const onDeclineRequest = form.handleSubmit((values) => {
+  const onDeclineRequest = async () => {
     if (!auth_id) {
       return toast.error('Unable to decline request: auth_id is missing ')
     }
 
+    // Guard: ensure organizations are loaded
+    if (!organizations || organizations.length === 0) {
+      return toast.error('Unable to decline request: Organizations not loaded')
+    }
+
+    const slugToUse = selectedOrgSlug || organization_slug || organizations[0]?.slug
```

Comment:
I'm thinking it would be better to stick with `values.selectedOrgSlug`. With all the fallbacks here, it might be easy to get into a bizarre state where the user has no idea what they're declining

Reply:
Good shout. I’ve just simplified/reverted the change so it does what you suggested.

Note: for similar reasons, I left the _Decline_ button enabled in the case that no organisation is selected. It shows the form validation error in the case a user clicks it without having selected an organisation:

> Please select an organization

This is instead of disabling the button until a organisation is selected. That would be technically neater but is cryptic. I.e. the user could reasonably have no idea why they can’t decline.